cli/azd/pkg/infra/provisioning/bicep/prompt.go

```diff
@@ -350,42 +350,61 @@ func (p *BicepProvider) promptForParameter(
 			return nil, fmt.Errorf("parameter '%s' has no allowed values defined", key)
 		}
 
+		// defaultOption enables running with --no-prompt, taking the default value. We use the first option as default.
+		defaultOption := options[0]
+		// user can override the default value with azd metadata
+		if azdMetadata.Default != nil {
```

Comment:
I understood from the PR description that:

```bicep
@metadata({azd: {
  default: 'foo'
}})
param fooParam string
```

is different than

```bicep
param fooParam string = 'foo'
```

Do you happen to know the context behind the intentional design separation? In an ideal world, I could wish for the Bicep defaulting case to be the de-facto implementation, but it seems that forces have pushed us down the first path. 

FWIW, I could see `defaultSelection` being slightly more descriptive than `default`, but I can see from the code diff that `default` was cemented in as part of the API previously.

Reply:
yeah, I initially named it `defaultSelection` but renamed it during PR review.  

The difference for one and the other is if the parameter selection should be presented to the user for awareness or not.
Bicep default prevents from any AZD-prompting, making users to possibly ignore available options, of forcing them to set ENV VARS or manually updating the bicep file to change things.

The defaultSelection helps templates to force a user interaction to at least accept the default proposed input for the template. 

We could make AZD to always prompt to confirm even default values... but then we could see feedback about too much prompting.

Reply:
Thanks for the explanation! I guess this does preserve Bicep semantics where default parameters would be accepted by default. And if I read `azd:{}` as "azd prompt metadata", this does all make a whole lot of sense.